### PR TITLE
sameer assignment

### PR DESCRIPTION
**Pull request by mohammedsameeralishaik**
Assignment

### CODE REVIEW

File: Assignments/Mohammed/README.md [hunk 1]
```diff
@@ -1 +1,149 @@
-# G7_Autocad
\ No newline at end of file
+--NAME: SHAIK MOHAMMED SAMEER ALI
+
+1. total number of employees
+A. SELECT count(empno) FROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:5]: ok

File: Assignments/Mohammed/README.md [hunk 2]
```diff
@@ -1 +1,149 @@
...
+2. total number of departments
+A. SELECT count(deptno) FROM dept;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:8]: ok

File: Assignments/Mohammed/README.md [hunk 3]
```diff
@@ -1 +1,149 @@
...
+3. all the employees
+A. SELECT ename FROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:11]: not ok.
select * from emp

File: Assignments/Mohammed/README.md [hunk 4]
```diff
@@ -1 +1,149 @@
...
+4. all the departments
+a.  SELECT dname FROM dept;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:14]: not ok
select * from dept;

File: Assignments/Mohammed/README.md [hunk 5]
```diff
@@ -1 +1,149 @@
...
+5. total salary paid for all employees.
+A. SELECT sum(sal) FROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:17]: ok

File: Assignments/Mohammed/README.md [hunk 6]
```diff
@@ -1 +1,149 @@
...
+6. total commission paid to all employees.
+A. SELECT sum(comm) FROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:20]: ok

File: Assignments/Mohammed/README.md [hunk 7]
```diff
@@ -1 +1,149 @@
...
+7. Which job titles of the employees get commission paid.
+A. SELECT distinct job FROM emp
+   WHERE comm is not null
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:24]: ok

File: Assignments/Mohammed/README.md [hunk 8]
```diff
@@ -1 +1,149 @@
...
+8. System date
+A. SELECT sysdate FROM dual;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:27]: ok

File: Assignments/Mohammed/README.md [hunk 9]
```diff
@@ -1 +1,149 @@
...
+9. Average salary paid to all employees.
+A. SELECT avg(saFROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:30]: not ok. syntax error

File: Assignments/Mohammed/README.md [hunk 10]
```diff
@@ -1 +1,149 @@
...
+10. How many employees are there in each department?
+A.  SELECT deptno, COUNT(empno) AS employee_count 
+    FROM Emp
+    GROUP BY deptno; 
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:35]: ok

File: Assignments/Mohammed/README.md [hunk 11]
```diff
@@ -1 +1,149 @@
...
+11. Total salary of the employees in each department
+A.  SELECT deptno, sum(sal) AS Total_sal  
+    FROM Emp
+    GROUP BY deptno; 
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:40]: ok

File: Assignments/Mohammed/README.md [hunk 12]
```diff
@@ -1 +1,149 @@
...
+12. How many employees are there under each job title?
+A.  SELECT job, count(job) AS groupof
+    FROM Emp
+    GROUP BY job; 
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:45]: ok

File: Assignments/Mohammed/README.md [hunk 13]
```diff
@@ -1 +1,149 @@
...
+13. Average salary paid for each job title.
+A.  SELECT job, avg(sal) AS avgof 
+    FROM Emp
+    GROUP BY job;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:50]: ok

File: Assignments/Mohammed/README.md [hunk 14]
```diff
@@ -1 +1,149 @@
...
+14. hire day, month, and year for each employee.
+A.  select ename, hiredate 
+    from emp
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:54]: not ok
select ename, to_char(hiredate,'dd'), to_char(hiredate,'mm'), to_char(hiredate,'yyyy') from emp

File: Assignments/Mohammed/README.md [hunk 15]
```diff
@@ -1 +1,149 @@
...
+15. Sort the employee department wise.
+A.  select * from emp
+    order by deptno;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:58]: ok

File: Assignments/Mohammed/README.md [hunk 16]
```diff
@@ -1 +1,149 @@
...
+16. Sort the employee based on their job titles.
+A.  SELECT empno, ename, job
+    FROM Emp
+    ORDER BY job;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:63]: ok

File: Assignments/Mohammed/README.md [hunk 17]
```diff
@@ -1 +1,149 @@
...
+17. Sort the employee based on descending order of their salaries.
+A.  SELECT empno, ename, job, sal
+    FROM Emp
+    ORDER BY sal desc;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:68]: ok

File: Assignments/Mohammed/README.md [hunk 18]
```diff
@@ -1 +1,149 @@
...
+18. Sort the employee ascending order of their department and descending order of their salary.
+A.  SELECT empno, ename, deptno, sal
+    FROM Emp
+    ORDER BY deptno asc , sal desc;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:73]: ok

File: Assignments/Mohammed/README.md [hunk 19]
```diff
@@ -1 +1,149 @@
...
+19. How many employees have their name with 6 characters.
+A.  SELECT COUNT(*) AS employee_count
+    FROM Emp
+    WHERE LENGTH(Ename) = 6;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:78]: ok

File: Assignments/Mohammed/README.md [hunk 20]
```diff
@@ -1 +1,149 @@
...
+20. Maximum and minimum salary paid.
+A.  SELECT MAX(sal) FROM emp;
+    SELECT MIN(sal) FROM emp;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:82]: not ok. one SQL needed
select max(sal), min(sal) from emp;

File: Assignments/Mohammed/README.md [hunk 21]
```diff
@@ -1 +1,149 @@
...
+21. Maximum, minimum, average and sum of salary paid under each department.
+A.  SELECT
+    deptno,
+    MAX(sal) AS max_sal,
+    MIN(sal) AS min_sal,
+    AVG(sal) AS avg_sal,
+    SUM(sal) AS total_sal
+    FROM Emp
+    GROUP BY deptno;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:92]: ok

File: Assignments/Mohammed/README.md [hunk 22]
```diff
@@ -1 +1,149 @@
...
+    AVG(sal) AS avg_sal,
+    SUM(sal) AS total_sal
+    FROM Emp
+    GROUP BY deptno;
+    
+22. Sort the employee based on their hire date.
+A.  SELECT empno, ename, hiredate
+    FROM Emp
+    ORDER BY hiredate;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:97]: ok

File: Assignments/Mohammed/README.md [hunk 23]
```diff
@@ -1 +1,149 @@
...
+23. Employee who joined latest
+A.  SELECT empno, ename, hiredate
+    FROM Emp
+    ORDER BY hiredate desc;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:102]: ok

File: Assignments/Mohammed/README.md [hunk 24]
```diff
@@ -1 +1,149 @@
...
+A.  SELECT empno, ename, to_char(hiredate,'yyyy')
+    FROM Emp
+    WHERE  to_char(hiredate,'yyyy') <= 1980
+    ORDER BY 1;
+    --or if we dont know year
+    SELECT empno, ename, to_char(hiredate, 'yyyy') AS hire_year
+    FROM Emp
+    ORDER BY hiredate ASC
+    FETCH FIRST 1 ROW ONLY;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:113]: ok

File: Assignments/Mohammed/README.md [hunk 25]
```diff
@@ -1 +1,149 @@
...
+25. Sort the employee based on their hire year (descending) and department (ascending)
+A.  SELECT empno, ename, deptno, hiredate
+    FROM Emp
+    ORDER BY  deptno asc , hiredate desc;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:118]: not ok. hireyear ordering needed
use to_char(hiredate,'YYYY') desc in the order by clause

File: Assignments/Mohammed/README.md [hunk 26]
```diff
@@ -1 +1,149 @@
...
+26. Employees who get salaries greater than or equal to the average salary of employees.
+A.  SELECT empno, ename, sal
+    FROM Emp
+    WHERE sal >= (SELECT AVG(sal) FROM Emp);
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:123]: ok

File: Assignments/Mohammed/README.md [hunk 27]
```diff
@@ -1 +1,149 @@
...
+27. Employees who get salary less than or equal to average salary of employees.
+A.  SELECT empno, ename, sal
+    FROM Emp
+    WHERE sal <= (SELECT AVG(sal) FROM Emp);
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:128]: ok

File: Assignments/Mohammed/README.md [hunk 28]
```diff
@@ -1 +1,149 @@
...
+28. Employees get salaries between 2000 and 4000.
+A.  SELECT empno, ename, job, sal, deptno
+    FROM emp
+    WHERE sal between 2000 and 4000;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:133]: ok

File: Assignments/Mohammed/README.md [hunk 29]
```diff
@@ -1 +1,149 @@
...
+    SELECT empno, ename, sal, job
+    FROM Emp
+    ORDER BY sal DESC
+    FETCH FIRST 1 ROW ONLY;
+    -- Lowest Salary
+    SELECT empno, ename, sal, job
+    FROM Emp
+    ORDER BY sal ASC
+    FETCH FIRST 1 ROW ONLY;
+
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:145]: ok
but combine as single query

File: Assignments/Mohammed/README.md [hunk 30]
```diff
@@ -1 +1,149 @@
...
+30. We need to celebrate the joining month of the employees. We plan to buy a gift for each of them. How many gifts I need to buy for next month?
+ A. SELECT empno, ename, hiredate
+    FROM Emp
+    WHERE EXTRACT(MONTH FROM hiredate) = 2;
```
Comment (raviproengage) [inline Assignments/Mohammed/README.md:149]: ok

**Review by raviproengage — COMMENTED** [review]
good submission. keep it up!

**Comment by raviproengage** [comment]
ok

**Comment by raviproengage** [comment]
ok

**Comment by raviproengage** [comment]
ok

File: Assignments/Mohammed/README.md (hunk 9 above)
Comment (mohammedsameeralishaik) [reply Assignments/Mohammed/README.md:30]: I will try to improve it sir, that the syntax error will not happen.

**Pull request closed without merging**